File: 算法/代码随想录算法PDF/01数组/367.有效的完全平方数.py
#
# @lc app=leetcode.cn id=367 lang=python3
#
# [367] 有效的完全平方数
#

# @lc code=start
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1: return True
        left = 1
        right = num // 2
        while left <= right:
            mid = left + (right - left) // 2
            if (mid ** 2 < num):
                left = mid + 1
            elif (mid ** 2 > num):
                right = mid - 1
            else:
                return True
        return False
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/367.有效的完全平方数.cpp
/*
 * @lc app=leetcode.cn id=367 lang=cpp
 *
 * [367] 有效的完全平方数
 */

// @lc code=start
class Solution {
public:
    bool isPerfectSquare(int num) {
        int item = 1;
        while (num > 0) {
            num -= item;
            item += 2;
        }
        return num == 0;
    }
};
// @lc code=end

